core/eval.go

```diff
@@ -87,6 +88,25 @@ func evalSET(args []string) []byte {
 				return Encode(errors.New("ERR value is not an integer or out of range"), false)
 			}
 			exDurationMs = exDurationSec * 1000
+
+		case "EXAT", "exat":
+			i++
+			if i == len(args) {
+				return Encode(errors.New("ERR syntax error"), false)
+			}
+
+			exAt, err := strconv.ParseInt(args[i], 10, 64)
+			if err != nil {
+				return Encode(errors.New("ERR value is not an integer or out of range"), false)
+			}
+
+			currentTime := time.Now().Unix()
+			if exAt <= currentTime {
+				return Encode(errors.New("ERR invalid EXAT time in the past"), false)
```

Comment:
This behaviour doesn't reflect the same what's done in redis. 
Redis does not reject the `EXAT` value if it is in the past. Instead, it sets the key and immediately expires it if the timestamp has already passed.
@JyotinderSingh @arpitbbhayani please correct me if I'm wrong here.